v2/gcs-to-splunk/src/test/java/com/google/cloud/teleport/v2/templates/GCSToSplunkTest.java

```diff
@@ -0,0 +1,311 @@
+import com.google.common.io.Resources;
+import org.apache.beam.sdk.coders.CoderRegistry;
+import org.apache.beam.sdk.coders.StringUtf8Coder;
+import org.apache.beam.sdk.io.splunk.SplunkEvent;
+import org.apache.beam.sdk.io.splunk.SplunkWriteError;
+import org.apache.beam.sdk.options.PipelineOptionsFactory;
+import org.apache.beam.sdk.testing.PAssert;
+import org.apache.beam.sdk.testing.TestPipeline;
+import org.apache.beam.sdk.transforms.Create;
+import org.apache.beam.sdk.transforms.Flatten;
+import org.apache.beam.sdk.transforms.ParDo;
+import org.apache.beam.sdk.values.PCollection;
+import org.apache.beam.sdk.values.PCollectionList;
+import org.apache.beam.sdk.values.PCollectionTuple;
+import org.apache.beam.vendor.guava.v26_0_jre.com.google.common.collect.ImmutableList;
+import org.junit.Rule;
+import org.junit.Test;
+
+/** Test cases for the {@link GCSToSplunk} class. */
+public class GCSToSplunkTest {
+  @Rule public final transient TestPipeline pipeline = TestPipeline.create();
+
+  private static final FailsafeElementCoder<String, String> FAILSAFE_ELEMENT_CODER =
+      FailsafeElementCoder.of(StringUtf8Coder.of(), StringUtf8Coder.of());
+
+  private static final String TRANSFORM_FILE_PATH =
+      Resources.getResource("elasticUdf.js").getPath();
+  private static final String NO_HEADER_CSV_FILE_PATH =
+      Resources.getResource("no_header.csv").getPath();
+  private static final String HEADER_CSV_FILE_PATH =
+      Resources.getResource("with_headers.csv").getPath();
+  private static final String JSON_SCHEMA_FILE_PATH =
+      Resources.getResource("testSchema.json").getPath();
+
+  /** Tests the {@link GCSToSplunk} pipeline using a Udf to parse the Csv. */
+  @Test
+  public void testGCSToSplunkUdf() {
+    // Arrange
+    final String stringifiedJsonRecord = "{\"id\":\"007\",\"state\":\"CA\",\"price\":26.23}";
+    SplunkEvent expectedSplunkEvent =
+        SplunkEvent.newBuilder().withEvent(stringifiedJsonRecord).create();
+
+    CoderRegistry coderRegistry = pipeline.getCoderRegistry();
+    coderRegistry.registerCoderForClass(SplunkEvent.class, SplunkEventCoder.of());
+    coderRegistry.registerCoderForType(
+        FAILSAFE_ELEMENT_CODER.getEncodedTypeDescriptor(), FAILSAFE_ELEMENT_CODER);
+
+    GCSToSplunkOptions options = PipelineOptionsFactory.create().as(GCSToSplunkOptions.class);
+
+    options.setJavascriptTextTransformGcsPath(TRANSFORM_FILE_PATH);
+    options.setJavascriptTextTransformFunctionName("transform");
+    options.setContainsHeaders(false);
+    options.setInputFileSpec(NO_HEADER_CSV_FILE_PATH);
+
+    // Act
+    PCollectionTuple readCsvOut = readFromCsvAndTransform(options);
+
+    // Assert
+    PAssert.that(readCsvOut.get(UDF_OUT))
+        .satisfies(
+            collection -> {
+              FailsafeElement element = collection.iterator().next();
+              assertThat(element.getPayload(), is(equalTo(stringifiedJsonRecord)));
+              return null;
+            });
+
+    // Act
```

Comment:
Tests should only have one Arrange, Act, and Assert block each. Please combine the two act and assert blocks into one. Same goes for other tests.